server/src-lib/Hasura/GraphQL/Execute.hs

```diff
@@ -747,44 +692,40 @@ arrayIndexText (ArrayIndex i) =
 
 -- | Produce a field from the nested field calls.
 fieldCallsToField ::
-     Maybe G.Alias
-  -> Map.HashMap G.Name AnnInpVal
+     Map.HashMap G.Name AnnInpVal
   -> Map.HashMap G.Variable G.ValueConst
   -> SelSet
+  -> Maybe G.Alias
   -> NonEmpty FieldCall
   -> Field
-fieldCallsToField mindexedAlias0 userProvidedArguments variables finalSelSet =
-  nest mindexedAlias0
+fieldCallsToField userProvidedArguments variables finalSelSet = nest
   where
-    nest mindexedAlias (fieldCall :| rest) =
+    nest mindexedAlias (FieldCall{..} :| rest) =
       Field
-        { _fAlias =
-            case mindexedAlias of
-              Just indexedAlias -> indexedAlias
-              Nothing           -> G.Alias (fcName fieldCall)
-        , _fName = fcName fieldCall
+        { _fAlias = fromMaybe (G.Alias fcName) mindexedAlias
+        , _fName = fcName
         , _fType = G.NamedType (G.Name "unknown_type")
         , _fArguments =
             let templatedArguments =
-                  createArguments variables (fcArguments fieldCall)
+                  createArguments variables fcArguments
```

Comment:
Let's discuss, first important thing is to give priority to `NonEmpty FieldCall` so we should swap the order of `Map.unionWith` on L716. See: https://github.com/tirumaraiselvan/graphql-engine/issues/28